Add documentation about OIDC and Federated Credentials (#37)
Azure/trusted-signing-action: README.md

> <p align="center">

Reviewer: Nit: `align` attribute on `p` [is obsolete](https://dev.w3.org/html5/spec-LC/obsolete.html).

```suggestion
<p style="text-align: center;">
```

Author: This doesn't seem to work correctly:
![image](https://github.com/Azure/trusted-signing-action/assets/59936622/c1900987-b67d-4256-930c-84150c64c5d5)

Reviewer: Oops, GitHub doesn't appear to support this. Ignore.

Author: No problem. Thank you for the review!

```

## Usage
Please review [Common error codes and mitigations](https://learn.microsoft.com/azure/trusted-signing/faq#common-error-codes-and-mitigations) before filing an issue.

### Authentication
Behind the scenes, the Action uses [DefaultAzureCredential](https://learn.microsoft.com/dotnet/api/azure.identity.defaultazurecredential?view=azure-dotnet) as the primary method of authentication to Azure. The [EnvironmentCredential](https://learn.microsoft.com/dotnet/api/azure.identity.environmentcredential?view=azure-dotnet) variables are exposed as inputs and then set to Action-scoped environment variables. Each credential type supported by `DefaultAzureCredential` can be disabled using the Action inputs.

> [!NOTE]

> > [Trusted Signing Certificate Profile Signer](https://learn.microsoft.com/azure/trusted-signing/concept-trusted-signing-resources-roles#supported-roles) role is required to successfully sign by using Trusted Signing

Reviewer: Nit
```suggestion
> [Trusted Signing Certificate Profile Signer](https://learn.microsoft.com/azure/trusted-signing/concept-trusted-signing-resources-roles#supported-roles) role is required to successfully sign with Trusted Signing
```
